docs/DeveloperGuide.md

```diff
@@ -897,60 +898,41 @@ Priorities: High (must have) - `***`, Medium (nice to have) - `**`, Low (unlikel
 
 ### Planned Enhancements
 
-#### User Interface
-
-* The user interface can show two lists views of both customer and delivery lists side by side to allow the user to see
-  both lists at the same time.
-
-#### Command Format
-
-* Prefixes can be shortened to allow the user to type less. For example, `--name` can be shortened to `-n`.
-* Potentially frequently used commands such as `customer list` and `delivery list` can be shortened to `cl` and `dl`
-  respectively.
-
-#### Error Messages
-
-* Error messages for dates would be more specific to allow the user to know what is wrong with the date that they
-  entered. For example, if a user enters 2023-02-30, the error message would be "Invalid date. The date entered does
-  not exist.".
-* Error messages for IDs should specify whether it is for the customer ID or for delivery ID as the current
-  implementation has some error messages that only specifies "ID" instead of "Customer ID" or "Delivery ID".
-* Error messages for numerical parameters that require a positive integer should throw more specific errors for negative
-  values.
-* Error messages would be improved in general to be more informative and instructive.
-
-#### User Accounts
-
-* Allow the user to create multiple accounts for different businesses.
-* Allow the user to create multiple accounts for the same business.
-
-#### Security
-
-* The whole JSON authentication file would be encrypted to prevent unauthorised access to the user's account instead of
-  only hashing the user password.
-* The whole delivery and customer data would be encrypted to prevent unauthorised access to the customer and delivery
-  data.
-
-#### ID Generation
+1. Currently, when a new customer or a new delivery is created, the ID is guaranteed to be unique among customers and
+   deliveries respectively but may not be generated from the first available ID. We plan to modify the ID generation so
+   that the ID of the new customer or new delivery would be the first available ID from 1 to the maximum integer
+   respectively. Thus, preventing the strange behaviour of the ID not starting from 1 when the data is removed
+   by `clear` or `delete account` or if the data is manually removed.
 
-* When a new customer or delivery is created, the ID of the customer or delivery would be the first available ID from 1
-  to the maximum integer in the list of customers or deliveries respectively.
-* ID of a new customer and delivery will reset to 1 for customers and deliveries if the user deletes all
-  customers and deliveries using the `clear` command.
-* ID of a new customer and delivery will reset to 1 for customers and deliveries if the user deletes his account.
+2. Currently, the `find` command for customer would allow special characters to be used to search for the name of a
+   customer. We plan to disallow special characters as a user cannot add special characters to the name of a customer in
+   the first place.
```

Comment:
Could we perhaps expand on the `find` command here as well, as it requires exact match of keywords, but perhaps in the future we can have more complex search, like fuzzy search or searching for phrases, e.g. i have 1 chocolate cake and 100 strawberry cakes and 100 chocolate buns, then searching "chocolate cake" should return the single instance.